doxia-core/src/main/java/org/apache/maven/doxia/index/IndexEntry.java

```diff
@@ -53,10 +56,41 @@ public class IndexEntry {
      */
     private List<IndexEntry> childEntries = new ArrayList<>();
 
+    public enum Type {
+        /**
+         * Used for unknown types but also for the root entry
+         */
+        UNKNOWN,
+        SECTION_1,
+        SECTION_2,
+        SECTION_3,
+        SECTION_4,
+        SECTION_5,
+        SECTION_6,
+        DEFINED_TERM,
+        FIGURE,
+        TABLE;
+
+        static Type fromSectionLevel(int level) {
+            if (level < SECTION_1.ordinal() || level > SECTION_6.ordinal()) {
+                throw new IllegalArgumentException("Level must be between " + SECTION_1.ordinal() + " and "
+                        + SECTION_6.ordinal() + " but is " + level);
+            }
+            return Arrays.stream(Type.values())
+                    .filter(t -> level == t.ordinal())
+                    .findAny()
+                    .orElseThrow(() -> new IllegalStateException("Could not find enum with ordinal " + level));
+        }
+    };
     /**
      * The type of the entry, one of the types defined by {@link IndexingSink}
      */
-    private final int type;
+    private final Type type;
+
+    /**
+     * System-dependent EOL.
+     */
+    private static final String EOL = System.getProperty("line.separator");
```

Comment:
Because you said in https://github.com/apache/maven-doxia/pull/180#discussion_r1428911493

>  there are tests which change the separator with try/finally

Not sure which tests you referred to and how they change it. If they change via reflection we must not consolidate the constants either...

Reply:
See: AbstractModuleTest.